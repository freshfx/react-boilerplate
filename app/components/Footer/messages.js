/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */
import {defineMessages} from 'react-intl'

export default defineMessages({
  authorMessage: {
    defaultMessage: `
      Made with love by {author}.
    `,
    id: 'boilerplate.components.Footer.author.message'
  },
  licenseMessage: {
    defaultMessage: 'This project is licensed under the MIT license.',
    id: 'boilerplate.components.Footer.license.message'
  }
})
