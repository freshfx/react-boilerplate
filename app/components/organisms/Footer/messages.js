/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */
import {defineMessages} from 'react-intl'

const scope = 'boilerplate.organisms.Footer'

export default defineMessages({
  authorMessage: {
    defaultMessage: `
      Made with love by {author}.
    `,
    id: `${scope}.author.message`
  },
  licenseMessage: {
    defaultMessage: 'This project is licensed under the MIT license.',
    id: `${scope}.license.message`
  }
})
