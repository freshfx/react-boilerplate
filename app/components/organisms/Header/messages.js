/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import {defineMessages} from 'react-intl'

const scope = 'boilerplate.organisms.Header'

export default defineMessages({
  features: {
    defaultMessage: 'Features',
    id: `${scope}.features`
  },
  home: {
    defaultMessage: 'Home',
    id: `${scope}.home`
  }
})
