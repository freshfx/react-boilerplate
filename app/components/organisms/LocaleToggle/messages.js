/*
 * LocaleToggle Messages
 *
 * This contains all the text for the LanguageToggle component.
 */
import {defineMessages} from 'react-intl'

const scope = 'boilerplate.organisms.LocaleToggle'

export default defineMessages({
  de: {
    defaultMessage: 'de',
    id: `${scope}.de`
  },
  en: {
    defaultMessage: 'en',
    id: `${scope}.en`
  }
})
