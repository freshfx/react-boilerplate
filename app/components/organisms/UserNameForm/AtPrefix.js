import styled from '@emotion/styled'

const AtPrefix = styled.span`
  color: black;
  margin-left: 0.4em;
`

export default AtPrefix
