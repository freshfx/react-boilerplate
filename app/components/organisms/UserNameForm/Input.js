import styled from '@emotion/styled'

const Input = styled.input`
  outline: none;
  border: none;
  border-bottom: 1px dotted #999;
  background-color: transparent;
`

export default Input
