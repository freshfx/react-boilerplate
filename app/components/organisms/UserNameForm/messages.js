import {defineMessages} from 'react-intl'

const scope = 'boilerplate.organisms.UserNameForm'

export default defineMessages({
  trymeAtPrefix: {
    defaultMessage: '@',
    id: `${scope}.tryme.atPrefix`
  },
  trymeMessage: {
    defaultMessage: 'Show Github repositories by',
    id: `${scope}.tryme.message`
  },
  trymePlaceholder: {
    defaultMessage: 'mxstbr',
    id: `${scope}.tryme.placeholder`
  }
})
