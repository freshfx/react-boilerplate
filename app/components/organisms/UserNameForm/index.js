import React, {useCallback} from 'react'
import PropTypes from 'prop-types'
import {useSelector} from 'react-redux'
import {FormattedMessage, useIntl} from 'react-intl'
import noop from 'lodash/noop'

import useDispatcher from 'hooks/useDispatcher'
import {actions, selectors} from 'modules/ui/username-form'

import AtPrefix from './AtPrefix'
import Form from './Form'
import Input from './Input'
import messages from './messages'

const UserNameForm = props => {
  const {formatMessage} = useIntl()
  const username = useSelector(selectors.selectUsername)
  const changeUsername = useDispatcher({action: actions.changeUsername})

  const onChange = useCallback(
    ({target: {value}}) => {
      changeUsername({username: value})
    },
    [changeUsername]
  )

  const onSubmit = useCallback(
    event => {
      event.preventDefault()
      props.onSubmit()
    },
    [props.onSubmit]
  )

  return (
    <Form onSubmit={onSubmit}>
      <label htmlFor="username">
        <FormattedMessage {...messages.trymeMessage} />
        <AtPrefix>
          <FormattedMessage {...messages.trymeAtPrefix} />
        </AtPrefix>
        <Input
          id="username"
          type="text"
          placeholder={formatMessage(messages.trymePlaceholder)}
          value={username}
          onChange={onChange}
        />
      </label>
    </Form>
  )
}

UserNameForm.propTypes = {
  onSubmit: PropTypes.func
}

UserNameForm.defaultProps = {
  onSubmit: noop
}

export default React.memo(UserNameForm)
