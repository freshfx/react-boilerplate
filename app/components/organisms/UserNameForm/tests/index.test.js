import React from 'react'
import {fireEvent} from '@testing-library/react'

import UsernameFormInjector from 'modules/ui/username-form/Injector'
import {actions} from 'modules/ui/username-form'
import render from 'utils/test-utils/custom-render'
import setupStore from 'utils/test-utils/setup-store'

import UserNameForm from '../index'
import messages from '../messages'

const {clearDispatch, options, store} = setupStore()

const renderComponent = (props = {}) =>
  render(<UserNameForm {...props} />, options)
const renderInjector = () => render(<UsernameFormInjector />, options)

describe('UserNameForm', () => {
  beforeAll(renderInjector)
  beforeEach(clearDispatch)

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  describe('events', () => {
    describe('change', () => {
      it('should dispatch a changeUsername action with the targets value', () => {
        const username = 'ffx'
        const {getByPlaceholderText} = renderComponent()
        fireEvent.change(
          getByPlaceholderText(messages.trymePlaceholder.defaultMessage),
          {target: {value: username}}
        )
        expect(store.dispatch).toHaveBeenCalledTimes(1)
        expect(store.dispatch).toHaveBeenCalledWith(
          actions.changeUsername({username})
        )
      })
    })

    describe('submit', () => {
      it('should call the onSubmit prop', () => {
        const onSubmit = jest.fn()
        const {container} = renderComponent({onSubmit})
        fireEvent.submit(container.firstChild)
        expect(onSubmit).toHaveBeenCalledTimes(1)
        expect(onSubmit).toHaveBeenCalledWith()
      })
    })
  })
})
