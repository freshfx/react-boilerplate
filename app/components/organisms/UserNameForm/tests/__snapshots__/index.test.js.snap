// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNameForm should match the snapshot 1`] = `
.emotion-0 {
  margin-bottom: 1em;
}

.emotion-2 {
  color: black;
  margin-left: 0.4em;
}

.emotion-4 {
  outline: none;
  border: none;
  border-bottom: 1px dotted #999;
  background-color: transparent;
}

<form
  class="emotion-0 emotion-1"
>
  <label
    for="username"
  >
    Show Github repositories by
    <span
      class="emotion-2 emotion-3"
    >
      @
    </span>
    <input
      class="emotion-4 emotion-5"
      id="username"
      placeholder="mxstbr"
      type="text"
      value=""
    />
  </label>
</form>
`;
